### DISCUSSION TITLE
[Warnings] Silence Vue warnings in DEV

### DISCUSSION THREAD

**Discussion by LaurentLord** [vuejs/rfcs#634] [General]
For compatibility reasons, I have to register my package twice (with different parameters). 

As a result, I get this error message: "Component "X" has already been registered in target app.".
However, I can't ignore these errors during development. In other words, my console is polluted by these multiple errors. 
Note that warnHandler doesn't catch them. 

Is it possible to provide the option of silencing some messages?

```js
// core/packages/runtime-core/src/apiCreateApp.ts
if (__DEV__ && context.components[name]) {
  warn(`Component "${name}" has already been registered in target app.`)
}
````
Source: https://github.com/vuejs/core/blob/ee4cd78a06e6aa92b12564e527d131d1064c2cd0/packages/runtime-core/src/apiCreateApp.ts#L278

Thanks

**Comment by dwjohnston** [comment]
+1

The scenario I have is that I have a component that has gets some data from `inject`. 

The injected value might not be provided and the components handles for this. 

If I render the component in a test, we see these warnings: 

```
[Vue warn]: injection "Symbol(injectionKeyLabel)" not found. 
```

This pollutes the test logs and is a distraction. We should be able to suppress this warning.

**Reply by jh-leong** [reply]
You can suppress this warning by providing a default value as the second parameter:

```js
inject(key, defaultValue)
```
